### PR TITLE
Add ListMember

### PR DESCRIPTION
**Pull request by MarcMorinville-Rewind**
This is not a "real" Klaviyo resource, it kind of dangles off List and amounts to an ID and an email. It's like the association between the Profile and the List.

Not very RESTful or AR-friendly, hence lots of custom code.

### CODE REVIEW

File: lib/klaviyo_api/resources/list.rb [hunk 1]
```diff
@@ -12,5 +12,32 @@ def find_single(scope, options)
...
+    # Returns an enumerator that knows how to transparently deal
+    # with Klaviyo's `marker` for pagination. Pages seem to be
+    # 1000 items.
+    #
+    # https://www.klaviyo.com/docs/api/v2/lists#get-members-all
+    def members(options = {})
+      CachingEnumerator.new do |yielder|
+        marker = nil
+
+        loop do
```
Comment (jhuff078) [inline lib/klaviyo_api/resources/list.rb]: A little concerned about hitting a rate limit and what's going to happen

File: lib/klaviyo_api/resources/list_member.rb [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    #
+    # https://www.klaviyo.com/docs/api/v2/lists#delete-members
+    def destroy
+      run_callbacks :destroy do
+        KlaviyoAPI::ListMember.delete self.email, prefix_options
+      end
+    end
+
+    def update
+      raise KlaviyoAPI::InvalidOperation.new 'Cannot update list members. You might be looking for delete and/or create.'
```
Comment (jhuff078) [inline lib/klaviyo_api/resources/list_member.rb]: Does this work?

I would expect
```
raise KlaviyoAPI::InvalidOperation, 'Cannot update list members. You might be looking for delete and/or create.'
```

File: lib/klaviyo_api/resources/list.rb (hunk 1 above)
Comment (smandegar) [reply lib/klaviyo_api/resources/list.rb]: I echo the concern without providing a solution

File: lib/klaviyo_api/resources/list_member.rb (hunk 2 above)
Comment (MarcMorinville-Rewind) [reply lib/klaviyo_api/resources/list_member.rb]: Yup it works! This test passes:

```
it 'raises InvalidOperation' do
      stub_request(:post, BASE_LIST_MEMBERS_URL)
        .to_return status: 200, body: load_fixture(:list_member)

      list_member = KlaviyoAPI::ListMember.create list_id: LIST_ID, email: 'someone@rewind.io'

      error = assert_raises KlaviyoAPI::InvalidOperation do
        list_member.update
      end

      assert_match 'Cannot update list members. You might be looking for delete and/or create.', error.message
    end
```

It's also the way the Honeybadger guys prefer to raise exceptions, for whatever that's worth. If we don't like it I can change it.

https://www.honeybadger.io/blog/a-beginner-s-guide-to-exceptions-in-ruby/

File: lib/klaviyo_api/resources/list.rb (hunk 1 above)
Comment (MarcMorinville-Rewind) [reply lib/klaviyo_api/resources/list.rb]: Hmm. Yes I see your point. I also don't have a solution. The question I pose to you is - is it the gem's responsibility to enforce rate limiting? If it's used as an enumerator with `next`, then users can slow their roll. If they do something like `count` then they'll get rate limited and know not to do that...

Comment (jhuff078) [reply lib/klaviyo_api/resources/list.rb]: The gem shouldn't be mitigating the rate limit but it has to provide the user the ability to do so in between calls

**Pull request merged**